Sources/Transport/TCPTransport.swift

```diff
@@ -131,7 +131,22 @@ public class TCPTransport: Transport {
             self?.delegate?.connectionChanged(state: .shouldReconnect(isBetter))
         }
         
-        conn.start(queue: queue)
+        start(conn, with: timeout)
+    }
+    
+    private func start(
+        _ connection: NWConnection,
+        with timeout: Double
+    ) {
+        let roundedTimeout = Int(timeout.rounded(.up))
+        connection.start(queue: queue)
+        
+        queue.asyncAfter(deadline: .now() + .seconds(roundedTimeout)) {
```

Comment:
What if we disconnect before timeout?

Reply:
Hm, then we better to use timer here so we can cancel it on disconnect.